Makefile target 'manpages' fails when executed in a network behind a proxy

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.

If you suspect your issue is a bug, please edit your issue description to
include the BUG REPORT INFORMATION shown below. If you fail to provide this
information within 7 days, we cannot debug your issue and will close it. We
will, however, reopen it if you later provide the information.

For more information about reporting issues, see
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#reporting-other-issues

---------------------------------------------------
GENERAL SUPPORT INFORMATION
---------------------------------------------------

The GitHub issue tracker is for bug reports and feature requests.
General support can be found at the following locations:

- Docker Support Forums - https://forums.docker.com
- IRC - irc.freenode.net #docker channel
- Post a question on StackOverflow, using the Docker tag

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Description**
<!--
Briefly describe the problem you are having in a few paragraphs.
-->
When trying to create a debian package (deb target), I had a few issues regarding the network proxy configuration. Some of them were fixed after including the appropriate proxy values using the `DOCKER_BUILD_ARGS` variable. Although, I was unable to successfully generate the package due to an error during man pages generation. 

Looking at the Makefile, I found that the 'manpages' target cannot be parameterized, which means that no proxy information will be available during the build and execution of that container. Due to the lack of proxy information, the apk command fails, resulting in the output that can be seen in the 'Describe the results you received' section.

**Steps to reproduce the issue:**
1. `make manpages` (ensure that you are in a network behind a proxy)

**Describe the results you received:**
```sh
svoboda@ecosystem-040:~/workspace/docker (master)$ make manpages
docker build -t docker-manpage-dev -f "man/Dockerfile" ./man
Sending build context to Docker daemon 785.4 kB
Step 1 : FROM alpine:3.4
 ---> baa5d63471ea
Step 2 : RUN apk add -U git go bash curl gcc musl-dev make
 ---> Running in 24033962ad4b
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/main/x86_64/APKINDEX.tar.gz
ERROR: http://dl-cdn.alpinelinux.org/alpine/v3.4/main: temporary error (try again later)
WARNING: Ignoring APKINDEX.167438ca.tar.gz: No such file or directory
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/community/x86_64/APKINDEX.tar.gz
ERROR: http://dl-cdn.alpinelinux.org/alpine/v3.4/community: temporary error (try again later)
WARNING: Ignoring APKINDEX.a2e6dac0.tar.gz: No such file or directory
ERROR: unsatisfiable constraints:
  bash (missing):
    required by: world[bash]
  curl (missing):
    required by: world[curl]
  gcc (missing):
    required by: world[gcc]
  git (missing):
    required by: world[git]
  go (missing):
    required by: world[go]
  make (missing):
    required by: world[make]
  musl-dev (missing):
    required by: world[musl-dev]
The command '/bin/sh -c apk add -U git go bash curl gcc musl-dev make' returned a non-zero code: 7
make: *** [manpages] Error 7
```

**Describe the results you expected:**
Response trimmed down to the important part.
```sh
svoboda@ecosystem-040:~/workspace/docker (master *)$ make DOCKER_BUILD_ARGS="--build-arg HTTP_PROXY=http://16.85.88.10:8088/ --build-arg HTTPS_PROXY=http://16.85.88.10:8088" manpages
docker build --build-arg HTTP_PROXY=http://16.85.88.10:8088/ --build-arg HTTPS_PROXY=http://16.85.88.10:8088 -t docker-manpage-dev -f "man/Dockerfile" ./man
Sending build context to Docker daemon 785.4 kB
Step 1 : FROM alpine:3.4
 ---> baa5d63471ea
Step 2 : RUN apk add -U git go bash curl gcc musl-dev make
 ---> Running in 7f202ebfa211
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/community/x86_64/APKINDEX.tar.gz
(1/30) Upgrading musl (1.1.14-r12 -> 1.1.14-r14)
(2/30) Installing ncurses-terminfo-base (6.0-r7)
(3/30) Installing ncurses-terminfo (6.0-r7)
(4/30) Installing ncurses-libs (6.0-r7)
(5/30) Installing readline (6.3.008-r4)
...
Step 13 : ENTRYPOINT man/generate.sh
 ---> Running in f05daed26038
 ---> 57854a2b823f
Removing intermediate container f05daed26038
Successfully built 57854a2b823f
docker run --rm \
		-v /home/svoboda/workspace/docker:/go/src/github.com/docker/docker/ \
		docker-manpage-dev
Generating man pages into ./man/man1
```

**Additional information you deem important (e.g. issue happens only occasionally):**
```diff
diff --git a/Makefile b/Makefile
index 8a54ef3..d9e7342 100644
--- a/Makefile
+++ b/Makefile
@@ -106,7 +106,7 @@ install: ## install the linux binaries
        KEEPBUNDLE=1 hack/make.sh install-binary
 
 manpages: ## Generate man pages from go source and markdown
-       docker build -t docker-manpage-dev -f "man/$(DOCKERFILE)" ./man
+       docker build ${DOCKER_BUILD_ARGS} -t docker-manpage-dev -f "man/$(DOCKERFILE)" ./man
        docker run --rm \
                -v $(PWD):/go/src/github.com/docker/docker/ \
                docker-manpage-dev
```
Applying this modification to the Makefile, the command `make DOCKER_BUILD_ARGS="--build-arg HTTP_PROXY=http://16.85.88.10:8088/ --build-arg HTTPS_PROXY=http://16.85.88.10:8088" manpages` completes successfully (as can be seen in the section 'Describe the results you expected').

**Output of `docker version`:**
```
Client:
Client:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        Wed Oct 26 21:44:32 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        Wed Oct 26 21:44:32 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**
```
Containers: 2
 Running: 0
 Paused: 0
 Stopped: 2
Images: 2
Server Version: 1.12.3
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 6
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null host bridge overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: apparmor
Kernel Version: 4.4.0-51-generic
Operating System: Ubuntu 14.04.5 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.797 GiB
Name: ecosystem-040.br.rdlabs.hpecorp.net
ID: LSX3:LZMH:VEHM:G7IB:6UB3:RW53:D7T6:4GAC:XHR7:6RH3:SYM6:VO6B
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Http Proxy: http://16.85.88.10:8088/
Https Proxy: http://16.85.88.10:8088/
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

--- a/Makefile
+++ b/Makefile
@@ -106,7 +106,7 @@ install: ## install the linux binaries
 	KEEPBUNDLE=1 hack/make.sh install-binary
 
 manpages: ## Generate man pages from go source and markdown
-	docker build -t docker-manpage-dev -f "man/$(DOCKERFILE)" ./man
+	docker build ${DOCKER_BUILD_ARGS} -t docker-manpage-dev -f "man/$(DOCKERFILE)" ./man
 	docker run --rm \
 		-v $(PWD):/go/src/github.com/docker/docker/ \
 		docker-manpage-dev

